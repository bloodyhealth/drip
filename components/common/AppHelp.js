import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, View } from 'react-native'

import AppText from '../common/app-text'

import { Containers, Spacing, Typography } from '../../styles'

const AppHelp = ({ text }) => (
  <View style={styles.container}>
    <AppText style={styles.accentPurple}>*</AppText>
    <AppText>{text}</AppText>
  </View>
)

AppHelp.propTypes = {
  text: PropTypes.string.isRequired,
}

const styles = StyleSheet.create({
  accentPurple: {
    ...Typography.accentPurple,
    alignSelf: 'flex-start',
    paddingRight: Spacing.base,
  },
  container: {
    ...Containers.rowContainer,
    padding: Spacing.base,
  },
})

export default AppHelp
