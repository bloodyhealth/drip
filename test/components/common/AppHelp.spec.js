import React from 'react'
import { render } from '@testing-library/react-native'
import AppHelp from '../../../components/common/AppHelp'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str, options) => {
      return str + (options ? JSON.stringify(options) : '')
    },
  }),
}))

describe('AppHelp screen', () => {
  test('when provided text, should render it', async () => {
    const text = 'Some help test'
    const { toJSON } = render(<AppHelp text={text} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
