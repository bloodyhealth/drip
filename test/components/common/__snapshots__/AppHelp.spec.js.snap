// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHelp screen when provided text, should render it 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 34.285714285714285,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#555",
          "fontFamily": "Jost-Book",
          "fontSize": 34.285714285714285,
        },
        Object {
          "alignSelf": "flex-start",
          "color": "#3A2671",
          "fontFamily": "Jost-Bold",
          "fontSize": 27.857142857142858,
          "paddingRight": 34.285714285714285,
          "textAlignVertical": "center",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    *
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#555",
          "fontFamily": "Jost-Book",
          "fontSize": 34.285714285714285,
        },
        undefined,
      ]
    }
  >
    Some help test
  </Text>
</View>
`;
